app.rb

```diff
@@ -0,0 +1,49 @@
+require 'sinatra'
+require 'pagy'
+require 'pagy/extras/headers'
+require 'active_support'
+require 'sinatra/activerecord'
+
+$LOAD_PATH.unshift settings.root + '/lib'
+
+require 'openapi3'
+
+class School < ActiveRecord::Base
+  def as_json
+    attrs = super.except('urn', 'coordinates')
+    {data: { id: urn, type: 'school', attributes: attrs }}
+  end
+end
+
+Pagy::DEFAULT[:items]  = 10
+
+class GIASApi < Sinatra::Base
+  include Pagy::Backend
+
+  before '/api/*' do
+    content_type 'application/json'
+  end
+
+  get '/' do
+    content_type 'application/json'
+
+    spec = YAML.load_file('config/gias_api_v1.yml').to_json
+  end
+
+  get '/api/schools' do
+    pagy, records = pagy(School.all)
+    pagy_headers_merge(pagy)
+
+    { data: records.as_json }.to_json
+  rescue ActiveRecord::RecordNotFound, Pagy::OverflowError, Pagy::VariableError
+    halt(404)
+  end
```

Comment:
Is pagination necessary? there are only ~25k schools and we're only holding a handful of fields for each

Reply:
I like this idea! will remove